castervoice/lib/ccr/core/text_manipulation.py

```diff
@@ -0,0 +1,254 @@
+                       dict(dictation="phrase", lease_ross="left_right"),
+                       cursor_behavior="texstudio"))]),
+                 rdescript="select chosen phrase"),
+        "grab lease [<number_of_lines_to_search>] <left_character>":
+            R(ContextAction(default=Function(text_manipulation_functions.select_phrase,
+            dict(left_character="phrase"), left_right="left",
+            cursor_behavior="standard"), actions=[(AppContext("texstudio"),
+            Function(text_manipulation_functions.select_phrase, dict(left_character="phrase"), left_right="left",
+            cursor_behavior="texstudio"))]),
+            rdescript="select chosen character to the left"),
+        "grab ross [<number_of_lines_to_search>] <right_character>":
+            R(ContextAction(default=Function(text_manipulation_functions.select_phrase, dict(right_character="phrase"), 
+            left_right="right", cursor_behavior="standard"), actions=[(AppContext("texstudio"),
+            Function(text_manipulation_functions.select_phrase, dict(right_character="phrase"), left_right="right",
+            cursor_behavior="texstudio"))]),
+            rdescript="select chosen character to the right"),
+        
+        "grab <lease_ross> [<number_of_lines_to_search>] until <dictation> ":
+            R(ContextAction(default=Function(text_manipulation_functions.select_until_phrase, 
+            dict(dictation="phrase", lease_ross="left_right"), 
+            cursor_behavior="standard"), actions=[(AppContext("texstudio"),
+            Function(text_manipulation_functions.select_until_phrase, dict(dictation="phrase", lease_ross="left_right"), 
+            cursor_behavior="texstudio"))]),
+                 rdescript="select until chosen phrase (inclusive)"),
+        "grab lease [<number_of_lines_to_search>] until <left_character>":
+            R(ContextAction(default=Function(text_manipulation_functions.select_until_phrase, dict(left_character="phrase"), 
+            left_right="left", cursor_behavior="standard"), actions=[(AppContext("texstudio"),
+            Function(text_manipulation_functions.select_until_phrase, dict(left_character="phrase"), 
+            left_right="left", cursor_behavior="texstudio"))]),
+            rdescript="select left until chosen character"),
+        "grab ross [<number_of_lines_to_search>] until  <right_character>":
+            R(ContextAction(default=Function(text_manipulation_functions.select_until_phrase, dict(right_character="phrase"), 
+            left_right="right", cursor_behavior="standard"),
+            actions=[(AppContext("texstudio"), Function(text_manipulation_functions.select_until_phrase,
+            dict(right_character="phrase"), left_right="right", cursor_behavior="texstudio"))]),
+            rdescript="select right until chosen character"),
+        "wipe <lease_ross> [<number_of_lines_to_search>] [<before_after>] <dictation>":
```

Comment:
The interface for wipe and grab should be identical because one just adds a backspace at the end right? Is wipe really necessary when I can just add "clear" after grab (I'm not really sure either way)? Should "select" be an option instead of "grab"? Also need to add "until" to wipe (if we keep it) and is there a reason to make "until" compulsory?

Reply:
There should also be a before/after for grab

Reply:
Right now `wipe` pastes the altered tex over the original text; it does not specifically select the text to be deleted. I think it's a little faster than the `grab until` commands and hence faster than "grab until clear". I think when I tried to use "select" in the spec, Dragon prioritized its native selection command.